src/gui/issueswidget.cpp

```diff
@@ -209,9 +214,35 @@ void IssuesWidget::slotOpenFile(QTreeWidgetItem *item, int)
 
 void IssuesWidget::slotProgressInfo(const QString &folder, const ProgressInfo &progress)
 {
-    if (progress.status() == ProgressInfo::Starting) {
-        // The sync is restarting, clean the old items
-        cleanItems(folder);
+    if (progress.status() == ProgressInfo::Reconcile) {
+        // Wipe all non-persistent entries - as well as the persistent ones
+        // in cases where a local discovery was done.
+        auto f = FolderMan::instance()->folder(folder);
+        if (!f)
+            return;
+        const auto &engine = f->syncEngine();
+        const auto style = engine.lastLocalDiscoveryStyle();
+        const auto &discoveryDirs = engine.currentLocalDiscoveryDirs();
+        cleanItems([&](QTreeWidgetItem *item) {
+            if (ProtocolItem::folderName(item) != folder)
+                return false;
+            if (style == LocalDiscoveryStyle::FilesystemOnly)
+                return true;
+            if (!persistsUntilLocalDiscovery(item))
+                return true;
+
+            // Definitely wipe the entry if the file no longer exists
+            if (!QFileInfo(f->path() + ProtocolItem::filePath(item)).exists())
```

Comment:
but maybe it only exists on the server!  (ignored on the server, or an error while downloading)

Anyway i don't think this is required.  If the file was deleted, this should trigger the filesystem watcher there anyway, wouldn't it?

Reply:
Yes, this check should be superfluous, but I was concerned about bugs that would make conflict file items stick around forever and adding the "does the file exist" check seemed like a good sanity check. But yes, the ignore could be remote and we have no way to tell. I'll remove it.